internal/storage/postgre.go

```diff
@@ -16,42 +24,195 @@ type Postgre struct {
 
 var _ Repository = (*Postgre)(nil)
 
+const dirMigrations = "internal/db/migrations"
+
 func NewPostgreBase(db *pgx.Conn, log *zap.Logger) *Postgre {
 	return &Postgre{
 		db:  db,
 		Log: log,
 	}
 }
 
+func Migrations(dsn string) error {
+	db, err := sql.Open("pgx", dsn)
+	if err != nil {
+		return err
+	}
+	defer func(db *sql.DB) {
+		err := db.Close()
+		if err != nil {
+			panic(err)
+		}
+	}(db)
+
+	if err := goose.SetDialect("postgres"); err != nil {
+		return err
+	}
+
+	return goose.Up(db, dirMigrations)
+}
+
 func (p *Postgre) Ping() error {
 	p.Log.Debug("Проверяем статус соединения с БД")
 	ctx, cancel := context.WithTimeout(context.Background(), 1*time.Second)
 	defer cancel()
 	if err := p.db.Ping(ctx); err != nil {
-		p.Log.Debug("Ошибка соединения с БД")
-		fmt.Println(err)
+		p.Log.Error("Ошибка соединения с БД")
 		return err
 	}
-	p.Log.Debug("Есть соединение с БД")
+	p.Log.Info("Есть соединение с БД")
 	return nil
 }
 
-func (p *Postgre) AddURL(data URLData) error {
-	//TODO implement me
-	panic("implement me1")
+func (p *Postgre) AddURL(ctx context.Context, data URLData) error {
+	p.Log.Debug("Добавляем URL в базу данных", zap.String("url", data.URL))
+	_, err := p.db.Exec(ctx, "INSERT INTO urls (id, url) VALUES ($1, $2)", data.ID, data.URL)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (p *Postgre) WriteBatchURL(ctx context.Context, b *ReqBatchURLs) (*ReqBatchURLs, error) {
+	tx, err := p.db.Begin(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, v := range *b {
+		// если данные не валидны, пропускаем текущую итерацию
+		if v.Err != nil {
+			continue
+		}
+
+		p.Log.Debug("Пытаемся добавить URL в БД", zap.String("url", v.URL), zap.String("id", v.ID))
+
+		_, err := tx.Exec(ctx, "INSERT INTO urls (id, url) VALUES ($1, $2)", v.ID, v.URL)
```

Comment:
It would be better if you use prepared statements

https://medium.com/@xuan11290/understand-mysql-prepared-statement-1eb1bda59f7b